### PR TITLE
Curb logging: change LogPhaseDuration from UserMessage into Diagnostics

### PR DESCRIPTION
**Pull request by almiliMSFT**

### CODE REVIEW

File: Public/Src/Engine/Processes/Tracing/Log.cs [hunk 1]
```diff
@@ -1020,14 +1021,15 @@ public abstract partial class Logger
...
+            EventTask = (ushort)Tasks.PipExecutor,
+            Message = "[{pipSemiStableHash}] Done with phase '{phaseName}' in {duration}.  {extraInfo}")]
+        public abstract void LogPhaseDuration(LoggingContext context, string pipSemiStableHash, string phaseName, string duration, string extraInfo);
 
-        public void LogSubPhaseDuration(LoggingContext context, string pipDescription, string phaseName, TimeSpan duration, string extraInfo = "")
+        public void LogSubPhaseDuration(LoggingContext context, Pip pip, SandboxedProcessFactory.SandboxedProcessCounters counter, TimeSpan duration, string extraInfo = "")
         {
-            LogPhaseDuration(context, pipDescription, phaseName, duration.ToString(), extraInfo);
+            SandboxedProcessFactory.Counters.AddToCounter(counter, duration);
+            LogPhaseDuration(context, pip.FormattedSemiStableHash, counter.ToString(), duration.ToString(), extraInfo);
```
Comment (mpysson) [inline Public/Src/Engine/Processes/Tracing/Log.cs]: >counter.ToString() [](start = 67, length = 18)

This used to have bad performance too under FullFramework. Luckily NetCore removed the hidden reflection :)

File: Public/Src/Engine/UnitTests/Scheduler/PipQueueTest.cs [hunk 2]
```diff
@@ -701,7 +701,7 @@ public class TestPipExecutor
                     PipExecutor.ReportExecutionResultOutputContent(
                         operationContext, 
                         environment, 
-                        pip.GetDescription(environment.Context), 
+                        pip.GetDescription(environment.Context),
```
Comment (almiliMSFT) [inline Public/Src/Engine/UnitTests/Scheduler/PipQueueTest.cs]:
```suggestion
                        pip.GetDescription(environment.Context), 
```

File: Public/Src/Engine/Processes/UnSandboxedProcess.cs [hunk 3]
```diff
@@ -393,7 +393,7 @@ protected void LogProcessState(string message)
             if (LoggingContext != null)
             {
                 string fullMessage = I($"Exited: {m_processExecutor?.ExitCompleted ?? false}, StdOut: {m_processExecutor?.StdOutCompleted ?? false}, StdErr: {m_processExecutor?.StdErrCompleted ?? false}, Reports: {ReportsCompleted()} :: {message}");
-                Tracing.Logger.Log.LogDetoursDebugMessage(LoggingContext, PipSemiStableHash, PipDescription, fullMessage);
+                Tracing.Logger.Log.LogDetoursDebugMessage(LoggingContext, PipSemiStableHash, string.Empty, fullMessage);
```
Comment (mpysson) [inline Public/Src/Engine/Processes/UnSandboxedProcess.cs]: >LogDetoursDebugMessage [](start = 35, length = 22)

how about removing the pipDescription parameter from the log event too?

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/UnSandboxedProcess.cs]: will do

File: Public/Src/Engine/Processes/SandboxedProcessMac.cs [hunk 4]
```diff
@@ -396,7 +396,7 @@ private async Task FeedStdInAsync(SandboxedProcessInfo info, [CanBeNull] string
 
             string line = I($"exec {info.FileName} {escapedArguments} {redirectedStdin}");
 
-            LogProcessState("Feeding stdin: " + line);
+            LogProcessState("Feeding stdin");
```
Comment (mpysson) [inline Public/Src/Engine/Processes/SandboxedProcessMac.cs]: >LogProcessState("Feeding stdin"); [](start = 12, length = 33)

Is this useful at this point?

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/SandboxedProcessMac.cs]: not really.  It takes a lot of space in the text log.  It's available in the XLG when needed.

**Pull request merged**